### PR TITLE
Use click as trigger for all popovers

### LINKED ISSUE #6272: Use buttons instead of adding tabindex=0 to elements
Grepping for `tabindex="0"` reveals a number of places where we're currently misusing non-interactive elements. While we have tried to improve accessibility by adding `role="button"` in many places, it's still best practice to just use a button if you want a button.

### PR DESCRIPTION
**Pull request by nwalters512**
Focus-triggered popovers have been a frequent source of strange UX issues, accessibility problems, and browser-specific bugs. Using clicks (and/or <kbd>Enter</kbd> and <kbd>Space</kbd>) to open and dismiss popovers should provide for a more reliable and consistent experience across the board.

Closes #6272.

> [!IMPORTANT]
> Please pay close attention to and check for the following while reviewing:
> - Ensure that all `<button>` tags have `type="button"`
> - Ensure that all opening/closing tags match

### CODE REVIEW

File: apps/prairielearn/elements/pl-drawing/pl-drawing.mustache [hunk 1]
```diff
@@ -6,7 +6,7 @@
 {{#parse_error}}
 <span>
     <span class="badge text-danger badge-invalid"><i class="fa fa-exclamation-triangle" aria-hidden="true"></i> Invalid</span>
-    <a tabindex="0" type="button" role="button" class="btn btn-sm btn-secondary small border ml-1" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" title="Format Error" data-content="{{parse_error}}"> Why <i class="fa fa-question-circle" aria-hidden="true"></i></a>
+    <button type="button" class="btn btn-sm btn-secondary small border ml-1" data-placement="auto"data-toggle="popover" data-html="true" title="Format Error" data-content="{{parse_error}}"> Why <i class="fa fa-question-circle" aria-hidden="true"></i></a>
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-drawing/pl-drawing.mustache:9]:
```suggestion
    <button type="button" class="btn btn-sm btn-secondary small border ml-1" data-placement="auto"data-toggle="popover" data-html="true" title="Format Error" data-content="{{parse_error}}"> Why <i class="fa fa-question-circle" aria-hidden="true"></i></button>
```

File: apps/prairielearn/elements/pl-units-input/pl-units-input.mustache [hunk 2]
```diff
@@ -77,29 +78,27 @@
...
 
     {{#missing_input}}
     {{#label}}<span>{{{label}}}</span>{{/label}}
     <span class="badge text-dark badge-missing-input">
         <i class="fa fa-question-circle" aria-hidden="true"></i> Missing Input
     </span>
-    <a href="javascript:void(0);" role="button" class="btn btn-sm btn-secondary small border"
-        data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true"
-        style="margin-left: 5px" title="Format Error" tabindex="0"
+    <button type="button" class="btn btn-sm btn-secondary small border"
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-units-input/pl-units-input.mustache:99]: Fix the closing tag here.

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 3]
```diff
@@ -169,19 +169,18 @@ export function InstructorAssessmentInstance({
                         <th>Fingerprint Changes</th>
                         <td colspan="2">
                           ${resLocals.assessment_instance.client_fingerprint_id_change_count}
-                          <a
-                            tabindex="0"
+                          <button
+                          type="button
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:173]:
```suggestion
                          type="button"
```

File: apps/prairielearn/elements/pl-units-input/pl-units-input.mustache (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/elements/pl-units-input/pl-units-input.mustache:99]: Resolved in 1928ef6.

File: apps/prairielearn/elements/pl-matrix-input/pl-matrix-input.mustache [hunk 4]
```diff
@@ -8,7 +8,7 @@
     <input name="{{name}}" type="text" class="form-control pl-matrix-input-input {{^display_append_span}}rounded-right{{/display_append_span}}" {{^editable}}disabled{{/editable}} {{#raw_submitted_answer}}value="{{raw_submitted_answer}}"{{/raw_submitted_answer}} placeholder="{{shortinfo}}" aria-labelledby="pl-matrix-input-{{uuid}}-label" />
     <div class="input-group-append">
         {{#show_info}}
-        <a role="button" class="btn btn-light border d-flex align-items-center" data-toggle="popover" data-html="true" title="Matrix" data-content="{{info}}" data-placement="auto" data-trigger="focus" tabindex="0">
+        <button class="btn btn-light border d-flex align-items-center" data-toggle="popover" data-html="true" title="Matrix" data-content="{{info}}" data-placement="auto">
```
Comment (mwest1066) [inline apps/prairielearn/elements/pl-matrix-input/pl-matrix-input.mustache:11]:
```suggestion
        <button type="button" class="btn btn-light border d-flex align-items-center" data-toggle="popover" data-html="true" title="Matrix" data-content="{{info}}" data-placement="auto">
```

File: apps/prairielearn/elements/pl-symbolic-input/pl-symbolic-input.mustache [hunk 5]
```diff
@@ -44,19 +44,16 @@
                 </span>
             {{/incorrect}}
             {{#parse_error}}
-                <a
-                    role="button"
+                <button
```
Comment (mwest1066) [inline apps/prairielearn/elements/pl-symbolic-input/pl-symbolic-input.mustache:47]:
```suggestion
                <button
                    type="button"
```

File: apps/prairielearn/src/components/QuestionScore.html.ts [hunk 6]
```diff
@@ -269,18 +269,16 @@ export function ExamQuestionStatus({
 
       ${(instance_question.allow_grade_left_ms ?? 0) > 0
         ? html`
-            <a
+            <button
```
Comment (mwest1066) [inline apps/prairielearn/src/components/QuestionScore.html.ts:272]:
```suggestion
            <button
              type="button"
```

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**